producers/http/handlers.go

```diff
@@ -136,11 +136,10 @@ func containerAppMetricHandler(p *producerImpl) http.HandlerFunc {
 			producers.AppMetricPrefix, cid,
 		}, producers.MetricNamespaceSep)
 
-		appMetrics, ok := p.store.Get(key)
-		if !ok {
-			httpLog.Errorf("/v0/containers/{id}/app/{metric-id} - not found in store: %s", key)
-			http.Error(w, "Key not found in store", http.StatusNoContent)
-			return
+		appMetrics, err := p.store.GetByRegex(key + ".*" + mid)
```

Comment:
You're right, this is bad logic. For what it's worth, this regex actually does fetch the appropriate keys because `store.GetByRegex` uses `regexp.MatchString`, which is confusingly a search rather than a match.